Classes/Classes/miniphonedir.cpp

```diff
@@ -0,0 +1,130 @@
+#include<iostream> 
+#include<map>
+
+class ContactData { 
+
+private: 
+
+    std::string _name;
+    std::string _last_name;
+    std::string _email;
+    std::string _phone_number;
+   
+public: 
+
+    ContactData(std::string name, std::string last_name, std::string email, std::string phone_number){
+        _name = name;
+        _phone_number = phone_number;
+        _email = email;
+        _last_name = last_name;
+    }
+    
+    std::string get_name(void){
+        return _name;
+    }
+}; 
+
+class Singleton_Organizer {
+    private:
+        Singleton_Organizer()
+        {
+            std::cout<<"Creating organizer" << std::endl;
+        }
+        std::map<int, ContactData*> map;
+        std::map<int, ContactData*>::iterator it;
+        int _id=0;
+
+    public: 
+        /*Getting singleton instance*/
+        static Singleton_Organizer& get_organizer(){
+            std::cout<<"Getting organizer from singleton"<<std::endl;
+            static Singleton_Organizer org;
+            return org;
+        }
+
+       void create_contact(std::string name, std::string last_name, std::string email, std::string phone_number) {
+            ContactData* contact = new ContactData(name,last_name,email,phone_number);  
+            map[_id] = contact;
+            ++_id;
+       }
+
+       bool delete_contact(int id) {
+            it = map.find(id);
+            if (it != map.end()){
+                map.erase (it);   
+                std::cout<< "ID: " << id << "was deleted" <<std::endl;
+            }
+            else
+            {
+                std::cout<<"Error: Element not found"<<std::endl;
+            }
+                    
+        
+       }
+
+       void display_contacts(void) {
```

Comment:
we need to come up with an sw design & architecture for the whole class, this case in particular we may select the way it's printed